Ruins/src/main/java/atomicstryker/ruins/common/RuinTemplateRule.java

```diff
@@ -754,7 +754,7 @@ private int getBlockNum(Random random)
     {
         // random selection using weights assigned in config file
         int blockIndex = 0;
-        for (int selector = random.nextInt(blockWeightsTotal); (selector -= blockWeights[blockIndex]) >= 0; ++blockIndex);
+        for (double selector = random.nextDouble()*blockWeightsTotal; (selector -= blockWeights[blockIndex]) >= 0; ++blockIndex);

```

Comment:
formatting (i know the file isn't fully formatted with my template i know ...)

Reply:
Again, not sure what the issue is. I'm guessing it's got something to do with whitespace...do you always put spaces around binary operators?

Reply:
"nextDouble() * blockWeightsTotal" the formatting template would do this, yes. I am a fan of mandatory formatting by the way, because when someone eventually formats everything the diff is illegible. It's a problem at work too.

Reply:
Got it. I don't have any strong stylistic preferences, myself, so I'll adapt to the environment.